### PR TITLE
fix #39

### LINKED ISSUE #39: Parser parses #IMPLIED use attributes as NORMAL (and vice versa)
I believe that there is a fundamental flaw in the `DTDParser`'s attribute use parsing. Using various test DTDs, it appears that attributes with `#IMPLIED` default declarations are assigned attribute use `USE_NORMAL`. Vice versa, attributes with a declared default value (and neither `#IMPLIED`, `#REQUIRED` nor `#FIXED` designation) do appear as `USE_IMPLIED`.

The following test case, I believe, demonstrates this issue with a very simple DTD:
```
import java.io.StringReader;
import java.util.HashMap;
import java.util.Map;
import com.sun.xml.dtdparser.DTDParser;
import com.sun.xml.dtdparser.DTDHandlerBase;
import com.sun.xml.dtdparser.DTDEventListener;
import org.junit.Test;
import org.xml.sax.InputSource;
import org.xml.sax.SAXException;
import static org.junit.Assert.assertEquals;

public class AttributeUseTest
{
  Map<String, Short> attributes = new HashMap<>();

  public AttributeUseTest() throws Exception
  {
    String dtd =
    """
      <!ELEMENT root EMPTY>
      <!ATTLIST root
        normal CDATA "default"
        implied CDATA #IMPLIED
        required CDATA #REQUIRED
        fixed CDATA #FIXED "default"
      >
    """;
    DTDEventListener handler = new DTDHandlerBase()
    {
      @Override
      public void attributeDecl(String element, String name, String type, String[] enums, short use, String defaultValue) throws SAXException
      {
          attributes.put(name, use);
      }
    };
    InputSource input = new InputSource(new StringReader(dtd));
    DTDParser parser = new DTDParser();
    parser.setDtdHandler(handler);
    parser.parse(input);
  }

  @Test
  public void testNormalAttribute()
  {
    assertEquals(DTDEventListener.USE_NORMAL, attributes.get("normal").shortValue());
  }

  @Test
  public void testImpliedAttribute()
  {
    assertEquals(DTDEventListener.USE_IMPLIED, attributes.get("implied").shortValue());
  }

  @Test
  public void testRequiredAttribute()
  {
      assertEquals(DTDEventListener.USE_REQUIRED, attributes.get("required").shortValue());
  }

  @Test
  public void testFixedAttribute()
  {
      assertEquals(DTDEventListener.USE_FIXED, attributes.get("fixed").shortValue());
  }
}
```
The test cases for `#REQUIRED` and `#FIXED` pass as expected, but the two other test cases both fail.

As I ran this in the debugger, I noticed some strange looking code in `DTDParser.maybeAttlistDecl()`:

https://github.com/eclipse-ee4j/jaxb-dtd-parser/blob/c9b1bb6b20eee9452561373faba867265353e1cc/dtd-parser/src/main/java/com/sun/xml/dtdparser/DTDParser.java#L1517-L1518

Intuitively, the  negation `!peek("#IMPLIED")` (instead of just `peek("#IMPLIED")`) seems incorrect here, since it would select `USE_IMPLIED` specifically if `#IMPLIED` was **not** present. This explains why attributes without any default declarations end up being recognized as implied.

Further down, `USE_NORMAL` is assigned as a fallback value:

https://github.com/eclipse-ee4j/jaxb-dtd-parser/blob/c9b1bb6b20eee9452561373faba867265353e1cc/dtd-parser/src/main/java/com/sun/xml/dtdparser/DTDParser.java#L1538-L1540

This explains why `#IMPLIED` attributes (which do not get recognized further up, due to the negation) are designated as `USE_NORMAL`.

### PR DESCRIPTION
**Pull request by antoniosanct**
@lukasj 
It seems to be an easy issue to fix

Regards,
Antonio.

### CODE REVIEW

File: dtd-parser/pom.xml [hunk 1]
```diff
@@ -294,6 +294,15 @@
         </plugins>
     </build>
 
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.12</version>
```
Comment (lukasj) [inline dtd-parser/pom.xml:301]: use the latest version instead (4.13.2, I think)

File: dtd-parser/src/main/java/com/sun/xml/dtdparser/DTDParser.java [hunk 2]
```diff
@@ -330,7 +329,6 @@ private void parseInternal(InputSource input)
                 fatal("P-003", null);
             }
         } catch (RuntimeException e) {
-            LOGGER.log(Level.SEVERE, "Internal DTD parser error.", e);
```
Comment (lukasj) [inline dtd-parser/src/main/java/com/sun/xml/dtdparser/DTDParser.java:333]: Why are you removing logger? It should stay there

**Review by lukasj — COMMENTED** [review]
new file needs a license header; the year in modified files needs to be updated to say 2023

File: dtd-parser/pom.xml (hunk 1 above)
Comment (antoniosanct) [reply dtd-parser/pom.xml]: Fixed in https://github.com/eclipse-ee4j/jaxb-dtd-parser/pull/41/commits/2ac4c77fa9702c42fc217fd0444fd1c1b64f422d

File: dtd-parser/src/main/java/com/sun/xml/dtdparser/DTDParser.java (hunk 2 above)
Comment (antoniosanct) [reply dtd-parser/src/main/java/com/sun/xml/dtdparser/DTDParser.java]: My mistake. I thought that it was unuseful. Restored in https://github.com/eclipse-ee4j/jaxb-dtd-parser/pull/41/commits/2ac4c77fa9702c42fc217fd0444fd1c1b64f422d

**Comment by antoniosanct** [comment]
> new file needs a license header; the year in modified files needs to be updated to say 2023

Fixed in https://github.com/eclipse-ee4j/jaxb-dtd-parser/pull/41/commits/2ac4c77fa9702c42fc217fd0444fd1c1b64f422d

**Review by lukasj — APPROVED** [review]
thanks!

**Pull request merged**